sdks/wasm/driver.c

```diff
@@ -149,7 +149,7 @@ mono_wasm_add_assembly (const char *name, const unsigned char *data, unsigned in
 		mono_register_symfile_for_assembly (new_name, data, size);
 		return;
 	}
-	WasmAssembly *entry = (WasmAssembly *)malloc(sizeof (MonoBundledAssembly));
+	WasmAssembly *entry = (WasmAssembly *)malloc(sizeof (WasmAssembly));
```

Comment:
We have `g_new` and `g_new0` for this.
I realize driver.c might not, but maybe copy them there if it doesn't.

Reply:
Seems like a good reason for `auto` too -- state the type just once.

Reply:
glib.h is not available there, there may be a reason @vargaz knows about. Also, it does seem like C99 supports `auto`, is it preferred to use C++11 instead ?

Reply:
Agreed -- we'd have to copy/paste/edit g_new and this is yet C.
They are just one line:

Searching for 'define g_new'...
```
C:\s\mono2\mono\eglib\glib.h(290):#define g_new(type,size)        ((type *) g_malloc (sizeof (type) * (size)))
C:\s\mono2\mono\eglib\glib.h(291):#define g_new0(type,size)       ((type *) g_malloc0 (sizeof (type)* (size)))
C:\s\mono2\mono\eglib\glib.h(292):#define g_newa(type,size)       ((type *) alloca (sizeof (type) * (size)))
```

Reply:
`g_malloc` also comes with glib, are there specifics in the `g_malloc` behavior, or a plain `malloc` is enough ?

Reply:
You'd change the macro to plain `malloc` most likely.
  It seems the lesser evil, than to do w/o `g_new`?
Or maybe provide `g_malloc` that `abort ()` upon failure.
Or maybe preferably, check for null in `driver.c` even if Mono often does not.

Reply:
This is a fine discussion for a separate PR.  I'm happy with landing this now

Reply:
https://github.com/mono/mono/pull/14890